account_multicurrency_revaluation/model/account.py

```diff
@@ -54,52 +47,52 @@ def _onchange_user_type_id(self):
             if rec.user_type_id.id in revaluation_accounts:
                 rec.currency_revaluation = True
 
-    def _revaluation_query(self, revaluation_date):
-
-        tables, where_clause, where_clause_params = self.env[
-            "account.move.line"
-        ]._query_get()
-
-        query = (
-            "with amount as ( SELECT aml.account_id, aml.partner_id, "
-            "aml.currency_id, aml.debit, aml.credit, aml.amount_currency "
-            "FROM account_move_line aml LEFT JOIN "
-            "account_partial_reconcile aprc ON (aml.balance < 0 "
-            "AND aml.id = aprc.credit_move_id) LEFT JOIN "
-            "account_move_line amlcf ON (aml.balance < 0 "
-            "AND aprc.debit_move_id = amlcf.id "
-            "AND amlcf.date < %s ) LEFT JOIN "
-            "account_partial_reconcile aprd ON (aml.balance > 0 "
-            "AND aml.id = aprd.debit_move_id) LEFT JOIN "
-            "account_move_line amldf ON (aml.balance > 0 "
-            "AND aprd.credit_move_id = amldf.id "
-            "AND amldf.date < %s ) "
-            "WHERE aml.account_id IN %s "
-            "AND aml.date <= %s "
-            "AND aml.currency_id IS NOT NULL "
-            "GROUP BY aml.id "
-            "HAVING aml.full_reconcile_id IS NULL "
-            "OR (MAX(amldf.id) IS NULL AND MAX(amlcf.id) IS NULL)"
-            ") SELECT account_id as id, partner_id, currency_id, "
-            + ", ".join(self._sql_mapping.values())
-            + " FROM amount "
-            + (("WHERE " + where_clause) if where_clause else " ")
-            + " GROUP BY account_id, currency_id, partner_id"
-        )
-
-        params = []
-        params.append(revaluation_date)
-        params.append(revaluation_date)
-        params.append(tuple(self.ids))
-        params.append(revaluation_date)
-        params += where_clause_params
-        return query, params
-
-    def compute_revaluations(self, revaluation_date):
+    def _revaluation_query(self, revaluation_date,company_id):
+
+# wrong is giving you a domain like 
+#         tables, where_clause, where_clause_params = self.env[
+#             "account.move.line"
+#         ]._query_get()
+        
+        query = f"""
```

Comment:
This change is unneeded and it's provoking a big diff. Please undo it.

Reply:
tables, where_clause, where_clause_params = self.env[
            "account.move.line"
        ]._query_get()
I commented in the code why I took out "params += where_clause_params "  - is giving you kind of odoo domain but is a sql querry so is useless

I know that odoo is telling not to use fstrings ( are python 3.6 and not 3.5); but this strings are making the code more readable. In future also odoo will do the same in their codes  and not to put  % % %  and some lists at the end of string

Reply:
It's not only that, it's the whole diff. Sorry, but this seems too much and we are not qualified for judging if it's correct. Wait for the merge and then propose your PR directly to OCA for others to judge.

Reply:
I took your code, I tested it and I found some problems and I changed them to be ok for me to use it. 
I sent back the code so you and others not to need to do the same.
The bugs were:
1. in report .py you are giving a data paramenter that is raising an error.
2. in your sql select you have to do a inner join with account.move to take there only lines that are posted   - now the code is revaluating also for canceled entries and also for draft.

without this 2 modification the code is should not be merged.
the rest that is modified in account.py   is just getting rid of unnecesary and unused functions variables and adding the company.

Reply:
Thanks for the explanation. @ernestotejeda please check what he says for being fixed on main PR